pkg/acpi/rsdp_linux.go

```diff
@@ -40,7 +40,7 @@ func GetRSDPEFI() (*RSDP, error) {
 		if start == "" {
 			continue
 		}
-		base, err := strconv.ParseUint(start, 0, 64)
+		base, err := strconv.ParseInt(start, 0, 63)
```

Comment:
s/63/64 it's still a 64bit value.

Reply:
but as an address and a seekable value it's 63, hence the change. Seek, for example, takes int64, and won't take anything more than 2^63-1 as a seekable absolute offset -- it's a negative number. 

offsets are a 63-bit number in any kernel I/O, so for  consistency through all this stuff I'd as soon leave it that way. It's not a pointer anyway ... it's a physical memory offset in the end.

e.g.
rminnich@xcpu:/tmp$ strace -e trace=pread64 ./a
pread64(3, 0x7ffcc90c9054, 1, -9223372036854775808) = -1 EINVAL (Invalid argument)
seek: Invalid argument
pread64(3, "\0", 1, 4611686018427387904) = 1
seek: Invalid argument
+++ exited with 0 +++
rminnich@xcpu:/tmp$ cat a.c
...
int main() {
	int buf[1], o;
	int fd = open("/dev/zero", 0);
	o = pread(fd, buf, 1, 1ULL<<63);  // EINVAL
	o = pread(fd, buf, 1, 1ULL<<62); // OK -- it' actually 1<<62 but hey ...
}

but if you're sure it should be 64 I'm not that tied to the argument :-)

Reply:
an int64 is still 64 bits, and you're only parsing 63 - right now, you can never parse an odd number, because the last bit is dropped

Reply:
actually I thought it got an error if the number was outside the range, let me check.
yeah, it does the right thing:
https://play.golang.org/p/v-2v6aCanTg

It won't drop the last bit, it errors on numbers greater than the bit width.

Reply:
https://play.golang.org/p/ki8e1ntnv6t is nicer.

Reply:
I see fair point